### PR TITLE
feat(action-menu)!: move to separate package

### PR DESCRIPTION
**Pull request by genaris**
Signed-off-by: Ariel Gentile <gentilester@gmail.com>

BREAKING CHANGE: action-menu module has been removed from the core and moved to a separate package. To integrate it in an Agent instance, it can be injected in constructor like this:
```ts
const agent = new Agent({
  config: { /* config */ },
  dependencies: agentDependencies,
  modules: {
    actionMenu: new ActionMenuModule(),
    /* other custom modules */
   }
})
```

Then, module API can be accessed in `agent.modules.actionMenu`.

### CODE REVIEW

**Review by TimoGlastra — APPROVED** [review]

File: packages/action-menu/src/__tests__/utils.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  })
+}
+
+export function getAgentConfig(name: string, extraConfig: Partial<InitConfig> = {}) {
+  const { config, agentDependencies } = getBaseConfig(name, extraConfig)
+  return new AgentConfig(config, agentDependencies)
+}
+
+const publicDidSeed = process.env.TEST_AGENT_PUBLIC_DID_SEED ?? '000000000000000000000000Trustee9'
+export function getBaseConfig(name: string, extraConfig: Partial<InitConfig> = {}) {
```
[the remark this answers is not in the document]
Comment (genaris) [reply packages/action-menu/src/__tests__/utils.ts:42]: Agree. Actually it seems that it was a copy-paste issue from question-answer, as I'm not using it. :-) Will update it in question-answer as well.

File: packages/action-menu/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "test": "jest"
+  },
+  "dependencies": {
+    "@aries-framework/core": "0.2.4",
+    "rxjs": "^7.2.0",
+    "class-transformer": "0.5.1",
+    "class-validator": "0.13.1"
+  },
+  "devDependencies": {
+    "@aries-framework/core": "0.2.4",
```
[the remark this answers is not in the document]
Comment (genaris) [reply packages/action-menu/package.json:34]: Yes. And regarding the version number, should it always match the version of AFJ core/node/react-native?

For the moment it's 0.2.4 but I guess we'll need to update to 0.3.0 for all packages in a further PR before merging to main.

File: packages/action-menu/README.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
     height="250px"
```
Comment (blu3beri) [inline packages/action-menu/README.md:6]: Not as a comment on your work, but we should probably export the dummyModule so other people can get started aswell :).

Comment (TimoGlastra) [reply packages/action-menu/README.md:6]: What do you mean exactly by this @blu3beri?

Comment (blu3beri) [reply packages/action-menu/README.md:6]: We have a dummyModule in AFJ explaining to people on how to create an extension module, the old way. If we update that everyone has a clear example to start modularising other modules and their custom modules that they might use.

Comment (TimoGlastra) [reply packages/action-menu/README.md:6]: Ah gotcha! Yes totally agree

**Pull request merged**